backend/src/v5/middleware/dataConverter/inputs/teamspaces/projects/modelSettings.js

```diff
@@ -0,0 +1,57 @@
+/**
+ *  Copyright (C) 2021 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+const { createResponseCode, templates } = require('../../../../../utils/responseCodes');
+const Yup = require('yup');
+const { respond } = require('../../../../../utils/responder');
+const { types } = require('../../../../../utils/helper/yup');
+
+const ModelSettings = {};
+
+ModelSettings.validateUpdateSettingsData = async (req, res, next) => {
+	const schema = Yup.object().shape({
+		name: types.strings.title,
+		desc: types.strings.blob,
+		surveyPoints: Yup.array()
+			.of(
+				Yup.object().shape({
+					position: types.position.required(),
+					latLong: Yup.array().of(Yup.number()).required()
```

Comment:
The test seems overly complicated? latLong should ever be 2 values, so just do this? 
```suggestion
					latLong: Yup.array().of(Yup.number()).length(2).required()
```